--- git status ---
On branch feature/imitation_rl
Your branch is up to date with 'origin/feature/imitation_rl'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	modified:   source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/agents/rsl_rl_ppo_cfg.py
	modified:   source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/action_cfg.py
	modified:   source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/commands_cfg.py
	modified:   source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/event_cfg.py
	modified:   source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/observation_cfg.py
	modified:   source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/termination_cfg.py
	modified:   source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/wbc_env_cfg.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/data/motions/csv/LAFAN1_Retargeting_Dataset/
	source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/data/motions/csv/convert_csv_to_npz.py
	source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/data/motions/npz/g1_jump_forward_traj.npz
	source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/data/motions/npz/jumps1_subject1.npz
	source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/data/motions/npz/play_npz.py
	unitree_rl_lab/

no changes added to commit (use "git add" and/or "git commit -a") 


--- git diff ---
diff --git a/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/agents/rsl_rl_ppo_cfg.py b/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/agents/rsl_rl_ppo_cfg.py
index ad1525a9..9c3e9c39 100644
--- a/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/agents/rsl_rl_ppo_cfg.py
+++ b/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/agents/rsl_rl_ppo_cfg.py
@@ -9,8 +9,8 @@ from isaaclab_rl.rsl_rl import RslRlOnPolicyRunnerCfg, RslRlPpoActorCriticCfg, R
 
 @configclass
 class BasePPORunnerCfg(RslRlOnPolicyRunnerCfg):
-    num_steps_per_env = 24
-    max_iterations = 30_000
+    num_steps_per_env = 24     # changed this parameter
+    max_iterations = 40_000
     save_interval = 500
     policy = RslRlPpoActorCriticCfg(
         init_noise_std=1.0,
diff --git a/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/action_cfg.py b/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/action_cfg.py
index abdae646..edaf2a8c 100644
--- a/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/action_cfg.py
+++ b/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/action_cfg.py
@@ -51,5 +51,5 @@ class G1ActionsCfg:
         ],
         scale=UNITREE_G1_29DOF_MIMIC_ACTION_SCALE, 
         use_default_offset=True, 
-        preserve_order=True,
+        preserve_order=True, # Maybe not neccesary as it is not on the Unitree Code
         )
\ No newline at end of file
diff --git a/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/commands_cfg.py b/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/commands_cfg.py
index 23686ef6..995f359b 100644
--- a/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/commands_cfg.py
+++ b/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/commands_cfg.py
@@ -51,5 +51,5 @@ class G1CommandsCfg:
             "right_elbow_link",
             "right_wrist_yaw_link",
         ],
-        motion_file=f"{MOTION_TRACKING_DATA_DIR}/motions/npz/g1_spinkick.npz", # <- change this for other motions
+        motion_file=f"{MOTION_TRACKING_DATA_DIR}/motions/npz/g1_jump_forward_traj.npz", # <- change this for other motions
     )
\ No newline at end of file
diff --git a/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/event_cfg.py b/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/event_cfg.py
index 37d715d4..2b273584 100644
--- a/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/event_cfg.py
+++ b/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/event_cfg.py
@@ -23,10 +23,11 @@ class G1EventCfg:
         func=mdp.randomize_rigid_body_material, # type: ignore
         mode="startup",
         params={
+            # Made some changes here
             "asset_cfg": SceneEntityCfg("robot", body_names=".*"),
-            "static_friction_range": (1.0, 1.0),
-            "dynamic_friction_range": (1.0, 1.0),
-            "restitution_range": (0.0, 0.0),
+            "static_friction_range": (0.3, 1.6),    # (1.0, 1.0)
+            "dynamic_friction_range": (0.3, 1.2),   # (1.0, 1.0)
+            "restitution_range": (0.0, 0.5),        # (0.0, 0.0)
             "num_buckets": 64,
         },
     )
diff --git a/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/observation_cfg.py b/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/observation_cfg.py
index 08a55457..bb41705b 100644
--- a/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/observation_cfg.py
+++ b/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/observation_cfg.py
@@ -22,6 +22,8 @@ class G1ObservationsCfg:
 
         # observation terms (order preserved)
         command = ObsTerm(func=mdp.generated_commands, params={"command_name": "motion"})
+
+        # What is this? Not too sure if this needs to be changed to the Unitree offical code.
         motion_anchor_pos_b = ObsTerm(
             func=wbc_mdp.motion_anchor_pos_b, params={"command_name": "motion"}, noise=Unoise(n_min=-0.25, n_max=0.25)
         )
diff --git a/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/termination_cfg.py b/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/termination_cfg.py
index 14d6e558..bf9a2169 100644
--- a/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/termination_cfg.py
+++ b/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/env_cfg/termination_cfg.py
@@ -43,7 +43,7 @@ class G1TerminationsCfg:
     )
 
     # -- safety terms
-    base_ang_vel_exceed = DoneTerm(
-        func=wbc_mdp.base_ang_vel_exceed,
-        params={"threshold": 500 * math.pi / 180.0},
-    )
\ No newline at end of file
+    # base_ang_vel_exceed = DoneTerm(
+    #    func=wbc_mdp.base_ang_vel_exceed,
+    #    params={"threshold": 500 * math.pi / 180.0},
+    # )
\ No newline at end of file
diff --git a/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/wbc_env_cfg.py b/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/wbc_env_cfg.py
index 97596bbb..5a5b56bf 100644
--- a/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/wbc_env_cfg.py
+++ b/source/isaaclab_tasks/isaaclab_tasks/manager_based/wbc/config/g1_29dof/wbc_env_cfg.py
@@ -13,6 +13,7 @@ from isaaclab.utils import configclass
 import isaaclab_tasks.manager_based.locomotion.velocity.mdp as mdp
 from isaaclab.envs import ManagerBasedRLEnvCfg
 from isaaclab_tasks.manager_based.locomotion.velocity.velocity_env_cfg import CurriculumCfg
+from isaaclab.envs.common import ViewerCfg
 
 ##
 # Pre-defined configs
@@ -31,7 +32,7 @@ from .env_cfg import (
 
 
 @configclass
-class G1WBCEnvCfg(ManagerBasedRLEnvCfg):
+class G1WBCEnvCfg_PLAY(ManagerBasedRLEnvCfg):
     """Configuration for the motion tracking environment."""
 
     # Scene settings
@@ -48,16 +49,23 @@ class G1WBCEnvCfg(ManagerBasedRLEnvCfg):
 
     def __post_init__(self):
         """Post initialization."""
-        # general settings
+        super().__post_init__()
+        self.episode_length_s = 20.0
         self.decimation = 4
-        self.episode_length_s = 10.0
-        # simulation settings
-        self.sim.dt = 0.005
-        self.sim.render_interval = self.decimation
-        self.sim.physics_material = self.scene.terrain.physics_material
-        self.sim.physx.gpu_max_rigid_patch_count = 10 * 2**15
-        # viewer settings
-        self.viewer.eye = (3.0, 3.0, 3.0)
+
+        self.terminations.anchor_pos = None # type: ignore
+        self.terminations.anchor_ori = None # type: ignore
+        self.terminations.ee_body_pos = None # type: ignore
+        self.terminations.base_ang_vel_exceed = None # type: ignore
+
+        self.viewer = ViewerCfg(
+            eye=(-0.0, -3.5, 0.2), 
+            lookat=(0.0, -0.0, 0.0),
+            resolution=(1920, 1080), 
+            origin_type="asset_root", 
+            asset_name="robot"
+        )
+    
 
         # disable debug vis
         # self.commands.motion.debug_vis = False
@@ -68,4 +76,5 @@ class G1WBCEnvCfg(ManagerBasedRLEnvCfg):
         # local_gui = carb_settings_iface.get("/app/window/enabled")
         # livestream_gui = carb_settings_iface.get("/app/livestream/enabled")
         # if not local_gui and not livestream_gui:
-        #     self.scene.dummy_robot = None
\ No newline at end of file
+        #     self.scene.dummy_robot = None
+G1WBCEnvCfg = G1WBCEnvCfg_PLAY
\ No newline at end of file